noether_examples/src/thickness_simulator_client.cpp

```diff
@@ -0,0 +1,182 @@
+#include <ros/ros.h>
+#include <ros/ros.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/point_types.h>
+#include <pcl/surface/vtk_smoothing/vtk_utils.h>
+#include <pcl/PolygonMesh.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <vtkPolyDataNormals.h>
+#include <vtkCellData.h>
+#include <vtkPointData.h>
+#include <vtk_viewer/vtk_utils.h>
+
+#include <tool_path_planner/raster_tool_path_planner.h>
+#include <tool_path_planner/utilities.h>
+#include <mesh_segmenter/mesh_segmenter.h>
+#include <actionlib/client/simple_action_client.h>
+#include <actionlib/client/terminal_state.h>
+#include <noether_msgs/ThickSimulatorAction.h>
+#include <noether_msgs/ToolRasterPath.h>
+
+//make modified meshes
+vtkSmartPointer<vtkPoints> createPlaneMod(unsigned int grid_size_x, unsigned int grid_size_y, vtk_viewer::plane_type type)
+{
+  // Create points on an XY grid with a sinusoidal Z component
+  vtkSmartPointer<vtkPoints> points = vtkSmartPointer<vtkPoints>::New();
+
+  for (unsigned int x = grid_size_x; x < 10; x++)
+  {
+    for (unsigned int y = 0; y < grid_size_y; y++)
+    {
+      switch (type)
+      {
+        case vtk_viewer::FLAT:
+          points->InsertNextPoint(x, y, vtkMath::Random(0.0, 0.001));
+          break;
+        case vtk_viewer::SINUSOIDAL_1D:
+          points->InsertNextPoint(x, y, sin(double(y) / 2.0) + vtkMath::Random(0.0, 0.001));
+          break;
+        case vtk_viewer::SINUSOIDAL_2D:
+        if(x==grid_size_x)
+        {
+           points->InsertNextPoint(x, y, 1.0 * cos(double(x) / 2.0) - 1.0 * sin(double(y) / 2.0) + vtkMath::Random(0.0, 0.001));
+        }
+        else
+        {
+           points->InsertNextPoint(x+0.25, y, 1.0 * cos(double(x) / 2.0) - 1.0 * sin(double(y) / 2.0) + vtkMath::Random(0.0, 0.001));
+        }
+        break;
+      }
+    }
+  }
+
+  // Add the grid points to a polydata object
+  vtkSmartPointer<vtkPolyData> polydata = vtkSmartPointer<vtkPolyData>::New();
+  polydata->SetPoints(points);
+
+  return points;
+}
+
+int main (int argc, char **argv)
```

Comment:
This would be a more compelling demo if it also allowed the user to add a mesh from a file in addition to those meshes you auto-generate.

Reply:
I will start working on it.